and-cli-deploy-azure-web-app.js

```diff
@@ -0,0 +1,184 @@
+                {
+                    echo.error("Error trying to add remote!");
+                    azure.logout();
+                    shell.exit(1);
+                }
+            }
+        },
+        pushToRemote() {
+            let pushCmd = `git push ${remote} ${branch}:master`;
+
+            if (force) {
+                pushCmd += " -f";
+            }
+
+            if (shell.exec(pushCmd).code !== 0) {
+                echo.error(" - Failed pushing to Web App remote");
+                azure.logout();
+                shell.exit(1);
+            }
+        },
+        description() {
+            return `Pushes indicated branch from the current git repo to indicated Azure Web App git repo, which then deploys to configured Azure Web App environment`;
+        },
+        async run() {
+            // Check system/command requirements
+            this.validateOrExit();
+
+            // Login to Azure
+            if (username != null) {
+                azure.login(username, secret);
+            } else {
+                azure.login(clientId, tenantId, secret);
+            }
+
+            this.createRemoteIfMissing();
+
+            this.pushToRemote();
+
+            // Logout from Azure
+            azure.logout();
+
+            echo.newLine();
+            echo.success("Application successfully deployed to Azure Web App!");
+        },
+        validateOrExit() {
+            const errors = [];
+
+            // Validate arguments
+            clientId = program.clientId;
+            tenantId = program.tenantId;
+            username = program.username;
+
+            const missingServicePrincipalArgs = (clientId == null || tenantId == null);
+
+            if (username == null && missingServicePrincipalArgs) {
+                errors.push("when --client-id or --tenant-id not provided, --username is required");
+            }
+
+            secret = program.secret;
+            if (secret == null) {
+                errors.push("--secret is required");
+            }
+
+            appName = program.appName;
+            if (appName == null) {
+                errors.push("--app-name is required");
+            }
+
+            resourceGroup = program.resourceGroup;
+            if (resourceGroup == null) {
+                errors.push("--resource-group is required");
+            }
+
+            branch = program.branch;
+            if (branch == null) {
+                errors.push("--branch is required");
+            }
+
+            remote = program.remote;
+            if (remote == null) {
+                errors.push("--remote is required");
+            }
+
+            if (program.force != null) {
+                force = true;
+            }
+
+            // Bail if up-front arguments are errored
+            if (errors.length > 0) {
+                echo.errors(errors);
+                shell.exit(1);
+            }
+
+            if (!shell.which("az")) {
```

Comment:
Thought: Consider moving this logic into shared azure module and updating both azure commands to use it

Reply:
Good catch, definitely a good idea.